language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionRouter.java

```diff
@@ -103,22 +108,24 @@ public static List<CodeAction> getAvailableCodeActions(CodeActionContext ctx) {
         // Get available diagnostics based code-actions
         ctx.diagnostics(ctx.filePath()).stream().
                 filter(diag -> PositionUtil
-                        .isWithinRange(position, PositionUtil.toRange(diag.location().lineRange()))
+                        .isRangeWithinRange(highlightedRange, PositionUtil.toRange(diag.location().lineRange()))
                 )
                 .forEach(diagnostic -> {
                     DiagBasedPositionDetails positionDetails = computePositionDetails(syntaxTree, diagnostic, ctx);
                     codeActionProvidersHolder.getActiveDiagnosticsBasedProviders(ctx)
-                                             .forEach(provider -> {
+                            .forEach(provider -> {
                                 try {
                                     // Check whether the code action request has been cancelled
                                     // in order to avoid unnecessary calculations
                                     ctx.checkCancelled();
 
-                                    if (!provider.validate(diagnostic, positionDetails, ctx)) {
+                                    DiagnosticBasedCodeActionProvider diagBasedProvider
+                                            = (DiagnosticBasedCodeActionProvider) provider;
```

Comment:
No need to cast. Let the holder return diag based providers only as I have commented earlier.